### PR TITLE
Cleans up naming in HTTP handlers

### LINKED ISSUE #1251: Clean up names
change the name of ssh key endpoint 
 - newCreateSSHKeyEndpoint
 - newDeleteSSHKey
 - newListSSHKeys
 - etc.

### PR DESCRIPTION
**Pull request by p0lyn0mial**
**What this PR does / why we need it**: simply cleans up naming in HTTP handlers.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #1251 

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: api/pkg/api/v1/types.go [hunk 1]
```diff
@@ -145,35 +144,36 @@ type AzureSize struct {
...
 	Fingerprint string `json:"fingerprint"`
 	PublicKey   string `json:"publicKey"`
 }
 
-// User represents an API user that is used for authentication.
-// Depreciated: use NewUser instead
-type User struct {
+// LegacyUser represents an API user that is used for authentication.
+// Depreciated: use User instead
+type LegacyUser struct {
```
Comment (p0lyn0mial) [inline api/pkg/api/v1/types.go:162]: I'm going to create a separate issue for removing `LegacyUser`

File: api/pkg/api/v1/types.go [hunk 2]
```diff
@@ -11,9 +11,9 @@ import (
 	cmdv1 "k8s.io/client-go/tools/clientcmd/api/v1"
 )
 
-// ObjectMeta is an object storing common metadata for persistable objects.
-// Deprecated: ObjectMeta is deprecated use NewObjectMeta instead.
-type ObjectMeta struct {
+// LegacyObjectMeta is an object storing common metadata for persistable objects.
+// Deprecated: LegacyObjectMeta is deprecated use ObjectMeta instead.
+type LegacyObjectMeta struct {
```
Comment (p0lyn0mial) [inline api/pkg/api/v1/types.go:16]: I'm going to create a separate issue for removing `LegacyObjectMeta`

**Review by maciaszczykm — APPROVED** [review]
LGTM!

File: api/pkg/api/v1/types.go (hunk 1 above)
Comment (p0lyn0mial) [reply api/pkg/api/v1/types.go:162]: https://app.zenhub.com/workspace/o/kubermatic/kubermatic/issues/2260

File: api/pkg/api/v1/types.go (hunk 2 above)
Comment (p0lyn0mial) [reply api/pkg/api/v1/types.go:16]: https://app.zenhub.com/workspace/o/kubermatic/kubermatic/issues/2261

**Pull request merged**